### PR TITLE
Handling case when response from GET is map only

### PR DESCRIPTION
**Pull request by subashd**

### CODE REVIEW

File: netscaler/config.go [hunk 1]
```diff
@@ -324,7 +325,15 @@ func (c *NitroClient) FindResource(resourceType string, resourceName string) (ma
 	}
 	resource := data[resourceType].([]interface{})[0] //only one resource obviously
```
Comment (chiradeep) [inline netscaler/config.go]: this line should be deleted?

File: netscaler/config.go [hunk 2]
```diff
@@ -324,7 +325,15 @@ func (c *NitroClient) FindResource(resourceType string, resourceName string) (ma
...
 
-	return resource.(map[string]interface{}), nil
+	rt := reflect.TypeOf(data[resourceType])
```
Comment (chiradeep) [inline netscaler/config.go]: How about:
```
//https://tour.golang.org/methods/16
switch result := data[resourceType].(type) {
   case map[string]interface{}:
       return result
   case []interface{}:
       return result[0]
   default:
       log.Printf("[WARN] go-nitro: FindResource Unable to determine type of response")
      return nil, fmt.Errorf("[INFO] go-nitro: FindResource: Unable to determine type of response")				
}

```

File: netscaler/config.go [hunk 3]
```diff
@@ -324,7 +325,15 @@ func (c *NitroClient) FindResource(resourceType string, resourceName string) (ma
...
+	case reflect.Map:
+		resource := data[resourceType]
+		return resource.(map[string]interface{}), nil
+	default:
+		resource := data[resourceType].([]interface{})[0] //only one resource obviously
+		return resource.(map[string]interface{}), nil
+	}
 }
 
 //FindAllResources finds all config objects of the supplied resource type and returns them in an array
```
Comment (chiradeep) [inline netscaler/config.go]: Do we not need similar logic here?

Comment (subashd) [reply netscaler/config.go]: As this functions return all config objects, I don't think changes are required here.

**Pull request merged**